src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -78,18 +78,19 @@ public function validate($entity, Constraint $constraint)
             }
         }
 
-        $class = $em->getClassMetadata(get_class($entity));
+        $entityClass = $constraint->entityClass ?: get_class($entity);
+        $class = $em->getClassMetadata($entityClass);
         /* @var $class \Doctrine\Common\Persistence\Mapping\ClassMetadata */
 
         $criteria = array();
         $hasNullValue = false;
 
-        foreach ($fields as $fieldName) {
-            if (!$class->hasField($fieldName) && !$class->hasAssociation($fieldName)) {
-                throw new ConstraintDefinitionException(sprintf('The field "%s" is not mapped by Doctrine, so it cannot be validated for uniqueness.', $fieldName));
+        foreach ($fields as $formField => $entityField) {
```

Comment:
`$sourceField`?

Reply:
Why not

Reply:
`$formField` => `$objectField`? because it is about the mapping field configuration between a DT**O** and the Entity.